### PR TITLE
Fix 101 bugs in various parts of the project.

### LINKED ISSUE #1985: RD had an error
I had an XLSM open with the VBE opened.
I opened an XLSX to look at it for a minute.
I closed the XLSX, then RD told me it had an issue, I should save & close & upload my log file, so here it is.
[rubberduck.log.txt](https://github.com/rubberduck-vba/Rubberduck/files/353023/rubberduck.log.txt)

I do not know what the error was.

### PR DESCRIPTION
**Pull request by Hosch250**
Close #1985 
Remove syncing the PE with the CE because it was incomplete and caused strange side effects.

### CODE REVIEW

**Comment by retailcoder** [comment]
101 bugs huh? :wink:

File: RetailCoder.VBE/API/ParserState.cs [hunk 1]
```diff
@@ -68,7 +68,7 @@ public void Initialize(VBE vbe)
 
             Func<IVBAPreprocessor> preprocessorFactory = () => new VBAPreprocessor(double.Parse(vbe.Version, CultureInfo.InvariantCulture));
             _attributeParser = new AttributeParser(new ModuleExporter(), preprocessorFactory);
-            _parser = new RubberduckParser(vbe, _state, _attributeParser, preprocessorFactory,
+            _parser = new RubberduckParser(_state, _attributeParser, preprocessorFactory,
```
Comment (retailcoder) [inline RetailCoder.VBE/API/ParserState.cs]: :+1: `dependencies--;`

File: RetailCoder.VBE/Navigation/CodeExplorer/CodeExplorerViewModel.cs [hunk 2]
```diff
@@ -87,17 +86,6 @@ public CodeExplorerItemViewModel SelectedItem
...
-                {
-                    var vbe = _selectedItem.GetSelectedDeclaration().Project.VBE;
-                    var project = vbe.VBProjects.Cast<VBProject>().FirstOrDefault(f => f.HelpFile == _selectedItem.GetSelectedDeclaration().Project.HelpFile);
-
-                    if (project != null)
-                    {
-                        vbe.ActiveVBProject = project;
-                    }
-                }
-
```
Comment (retailcoder) [inline RetailCoder.VBE/Navigation/CodeExplorer/CodeExplorerViewModel.cs]: Are we not setting the `vbe.ActiveVBProject` upon selection changing?

File: Rubberduck.Parsing/VBA/RubberduckParserState.cs [hunk 3]
```diff
@@ -209,49 +209,23 @@ public void OnStatusMessageUpdate(string message)
...
-                if (!_projects.ContainsKey(projectId))
-                {
-                    _projects.Add(projectId, project);
-                }
-
-                foreach (VBComponent component in project.VBComponents)
-                {
-                    _moduleStates.TryAdd(new QualifiedModuleName(component), new ModuleState(ParserState.Pending));
-                }
+                _projects.Add(project.HelpFile, project);
```
Comment (retailcoder) [inline Rubberduck.Parsing/VBA/RubberduckParserState.cs]: I can't shake the feeling that `project.HelpFile` shouldn't be referred to directly. I mean we gave it another meaning entirely, there should be a tiny little level of abstraction here, just so we don't have 200 places to tweak if our strategy for a ProjectID ever needs to change. Thoughts?

File: RetailCoder.VBE/Navigation/CodeExplorer/CodeExplorerViewModel.cs (hunk 2 above)
Comment (Hosch250) [inline RetailCoder.VBE/Navigation/CodeExplorer/CodeExplorerViewModel.cs]: I removed that for a couple reasons.

A) I was only setting that when the project was selected in the VBE.  Nothing else was being synced, and it wasn't even doing what we intended.
B) It was created a QualifiedModuleName somewhere, causing a slight flicker as the VBE quickly swapped the active project.  Actually, I need to make that a property so we do this as little as possible.

File: Rubberduck.Parsing/VBA/RubberduckParserState.cs (hunk 3 above)
Comment (Hosch250) [inline Rubberduck.Parsing/VBA/RubberduckParserState.cs]: Makes sense.  Maybe add an extension method `GetProjectId`?  Would extension properties be nice?

File: RetailCoder.VBE/Navigation/CodeExplorer/CodeExplorerViewModel.cs (hunk 2 above)
Comment (retailcoder) [inline RetailCoder.VBE/Navigation/CodeExplorer/CodeExplorerViewModel.cs]: So... the *properties* toolwindow doesn't reflect the CE selection?

Comment (Hosch250) [inline RetailCoder.VBE/Navigation/CodeExplorer/CodeExplorerViewModel.cs]: Actually, the VBE somehow (or it did, anyway) update the selected project when a node was right-clicked.  It won't be hard to set the selected project in the execute for that command, anyway.

File: Rubberduck.Parsing/VBA/RubberduckParserState.cs (hunk 3 above)
Comment (retailcoder) [inline Rubberduck.Parsing/VBA/RubberduckParserState.cs]: lol, an extension method would do it. as for extension properties... we're not on C# 6, so even if you implement it and it ships with C# 8... :wink:

Comment (Hosch250) [inline Rubberduck.Parsing/VBA/RubberduckParserState.cs]: Actually, the Roslyn team doesn't accept major PR's from non-insiders anyway.

**Pull request merged**